### DISCUSSION TITLE
Are Windows runners supported?

### DISCUSSION THREAD

**Discussion by DrShushen** [community/community#160698] [Actions] [answered]
Hello all,

A Kubernetes/ARC beginner here.

I wanted to understand whether Windows runners are currently supported?

I see that, for instance, here there is information on Windows runners:
https://github.com/actions/actions-runner-controller/blob/master/docs/configuring-windows-runners.md

But in a recent comment on the video here I can see that it's stated that Windows runners are not supported:
https://www.youtube.com/watch?v=_F5ocPrv6io&ab_channel=glich.stream

Could you please clarify?

**Comment by roshitho** [comment]
I would also like to know if there is support for windows runner. Any guidance on the same  would help.

**Comment by Link-** [comment, marked as answer]
There will be no official support for windows runners with ARC. That might change in the future, but there are no concrete plans for it. 
Attempts were made by different entities that were successful. You're free to experiment yourself with this.

**Reply by JohnLBergqvist** [reply]
I'm new to the project, so please forgive me if i'm missing something obvious, but if there's no official support for Windows runners, why is this document here? https://github.com/actions/actions-runner-controller/blob/master/docs/configuring-windows-runners.md

**Reply by damienomurchu** [reply]
My team and I have been successfully running ARC v1 with windows runners for the past 6 months or so after following https://github.com/actions/actions-runner-controller/blob/master/docs/configuring-windows-runners.md

@Link- while I understand the current situation is not to support Windows runners with ARC v2 - is that due to potential technical issues, or rather just a decision to limit the ARC support overhead to linux runners due to bandwidth/ resources, etc?

**Reply by Link-** [reply]
@JohnLBergqvist This is the documentation for the legacy ARC modes which are supported by the community only.

**Reply by Link-** [reply]
@damienomurchu They will introduce additional technical overhead which we're not equipped to handle at the moment. We might revise this in the future

**Reply by damienomurchu** [reply]
Thanks @Link-, Windows runners work quite well on ARC v1 (on EKS) for our organisation, but moving to a GitHub-supported version of ARC for Linux and Window runners would be our preferred path. 

We'll continue to monitor the situation with Windows support for ARC v2 should things change in the future.

**Reply by Link-** [reply]
@damienomurchu I'm not aware of technical blockers for windows runners, so you can experiment with them. The only caveat is that if an issue arises, we don't support that path.

**Comment by macux** [comment]
@Link- we're using Windows runners successfully and for us it's an important requirement. Do you see any specific potential issues for teams like us given the lack of "official" support here?

**Reply by roshitho** [reply]
@macux , could you please advise if you are following - https://github.com/actions/actions-runner-controller/blob/master/docs/configuring-windows-runners.md to get windows runner working. Also can you please let me know which version of arc you are using and if any pointer/sample to your base runner image. Thanks

**Reply by Link-** [reply]
If it's working then it's working :) I don't foresee any problems given the current circumstances

**Reply by madanmishra19** [reply]
i was facing the problem.

**Comment by macux** [comment]
@roshitho yes we did follow that link and we're using the legacy version but have tested Windows runners on the new version too. I think on the main github runner repo there's a Dockerfile which was used as the basis for our Windows image, hth.

There's also some more info in [this issue](https://github.com/actions/actions-runner-controller/issues/2652) I created (which has now been fixed).

**Reply by jianliang00** [reply]
Could you share your Dockerfile used on the new version? I'm struggling in setting up a windows runner using the new github runner scale set.

**Comment by DrShushen** [comment]
@Link-, thank you for the response. What would be the best venue for the users to express interest in this feature (official Windows runner support), so that the intertest in this can be gaged, and the likelihood of this being picked up in future improved?

**Reply by Link-** [reply]
This is the best venue: https://github.com/orgs/community/discussions/categories/actions - it's also possible through your GitHub account manager / solutions engineer / services delivery engineer

**Comment by adiroiban** [comment]
While Windows are not supported, I think that the current ARC controller code is good enought to spin WIndows pods.

Here is an example of docker file + runner-set-values.yaml for `helm` deployment.

https://gist.github.com/adiroiban/cd3ccdb9b28e949123bb51914de8d96b

The biggest challenge is getting a Windows node into your k8s cluster and then building a container image good enought to run your tests.

**Reply by jianliang00** [reply]
Thanks for your sharing. I have tried the configuration, but I'm blocking on finding a working dynamic local path provisoner on Windows now. Would you mind to share some more infomation about how to configure the PV provisoner?

**Reply by bgrainger** [reply]
Thanks @adiroiban; that was helpful. In case it helps anyone else, I wrote up a description of how we're autoscaling Windows runners on AKS using ARC v2: https://faithlife.codes/blog/2025/10/windows-on-actions-runner-controller/
